torchci/pages/api/drci/drci.ts

```diff
@@ -40,26 +51,104 @@ export async function updateDrciComments(octokit: Octokit, prNumber?: string) {
         NUM_MINUTES + ""
     );
 
-    const workflowsByPR = reorganizeWorkflows(recentWorkflows);
+    const workflowsByPR = await reorganizeWorkflows(octokit, recentWorkflows);
     const sevs = getActiveSEVs(await fetchIssuesByLabel("ci: sev"));
+    const masterFlakyJobs = await getFlakyJobs();
+    const baseCommitJobs = await getBaseCommitJobs(workflowsByPR);
 
     for (const [pr_number, pr_info] of workflowsByPR) {
-        const { pending, failedJobs } = getWorkflowJobsStatuses(pr_info);
+        const { pending, failedJobs, flakyJobs, brokenTrunkJobs } =
+          getWorkflowJobsStatuses(
+            pr_info,
+            masterFlakyJobs,
+            baseCommitJobs.get(pr_info.merge_base) || new Map()
+          );
 
-        const failureInfo = constructResultsComment(pending, failedJobs, pr_info.head_sha);
-        const comment = formDrciComment(pr_number, failureInfo, formDrciSevBody(sevs));
+        const failureInfo = constructResultsComment(
+          pending,
+          failedJobs,
+          flakyJobs,
+          brokenTrunkJobs,
+          pr_info.head_sha,
+          pr_info.merge_base
+        );
+        const comment = formDrciComment(
+          pr_number,
+          failureInfo,
+          formDrciSevBody(sevs)
+        );
 
         await updateCommentWithWorkflow(octokit, pr_info, comment);
     }
+    console.log("done")
+}
+
+
+async function getBaseCommitJobs(
+  workflowsByPR: Map<number, PRandJobs>
+): Promise<Map<string, Map<string, RecentWorkflowsData>>> {
+  // get merge base shas
+  let baseShas = [];
+  for (const [_, pr_info] of workflowsByPR) {
+    baseShas.push(pr_info.merge_base);
+  }
+
+  // fetch failing jobs on those shas
+  const commitFailedJobsQueryResult = await fetchCommitFailedJobs(baseShas);
+
+  // reorganize into a map of sha -> name -> data
+  const jobsBySha = new Map();
+  for (const job of commitFailedJobsQueryResult) {
+    if (!jobsBySha.has(job.head_sha)) {
+      jobsBySha.set(job.head_sha, new Map());
+    }
+    const existing_job = jobsBySha.get(job.head_sha).get(job.name!);
+    if (!existing_job || existing_job.id < job.id!) {
+      // if rerun, choose the job with the larger id as that is more recent
+      jobsBySha.get(job.head_sha).set(job.name, job);
+    }
+  }
+  return jobsBySha;
+}
+
+async function getFlakyJobs(): Promise<FlakyJob[]> {
+  const urlkey =
+    "https://raw.githubusercontent.com/pytorch/test-infra/generated-stats/stats/rules.json";
+
+  const result = await urllib.request(urlkey);
```

Comment:
Might be worth adding a retry policy here

Reply:
im having trouble getting urllib's retry option (maybe our version is too old?) so im going to leave this for later, worst thing that happens is we dont get a json and just return an empty array => dont get the classification, which is conservative so im willing to let it go

Reply:
An option would be to create our own retry loop here, but returning an empty array is fine too.